From aba48e55cfe37aff1ef04cabdd6d019daa47de8a Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 08 Feb 2021 20:19:46 +0200
Subject: [PATCH] [SW-20154] make security keys handling platform specific

In Gaudi2, with the recent changes in specs, there is
provision for only a total of 3 keys(including one Habana
development key). Hence this recent change in specs, will
cause a compile error for Gaudi2. To fix this, make
this function platform specific. The actual implementation
is added in the next patches.

Change-Id: Ic0044f47abd8503edb0ff6a5afa5fab0ba0bce1c
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index f5a589a..0c6a4c1 100644
--- a/common/security.c
+++ b/common/security.c
@@ -38,31 +38,6 @@
 		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
-uint32_t security_key_soc_location_offset(uint32_t hash_key_index)
-{
-	uint32_t ret;
-	uint8_t key_soc_base_index;
-	uint8_t key_soc_location_index;
-	uint32_t key_soc_location[] = {
-		mmPSOC_SECURITY_HABANA_KEY_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_0_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_1_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_2_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_3_0
-	};
-
-	/* select which index to choose (haban, customer 0-4) */
-	key_soc_base_index = hash_key_index / SECURITY_DWORD_PER_KEY;
-	/* key is 256bit, 32bit * 8... which of the 8 location is requested */
-	key_soc_location_index = hash_key_index % SECURITY_DWORD_PER_KEY;
-
-	/* get base address of selected key index */
-	ret = key_soc_location[key_soc_base_index];
-	/* add to base address of key the spicifice location of currect DWORD */
-	ret += (key_soc_location_index * sizeof(uint32_t));
-	return ret;
-}
-
 void security_spi_dma_mmu_bypass(void)
 {
 	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
diff --git a/gaudi2_arc/security.c b/gaudi2_arc/security.c
index 2a0b974..9afb905 100644
--- a/gaudi2_arc/security.c
+++ b/gaudi2_arc/security.c
@@ -87,3 +87,10 @@
 	security_dcon_if0_rr_hbw_config(spi_en);
 	security_dcon_if1_rr_hbw_config(spi_en);
 }
+
+uint32_t security_key_soc_location_offset(
+			__attribute__ ((unused)) uint32_t hash_key_index)
+{
+	/* TODO: will be added in next patch */
+	return 0;
+}
diff --git a/goya2_arc/security.c b/goya2_arc/security.c
index d9c290d..5290a38 100644
--- a/goya2_arc/security.c
+++ b/goya2_arc/security.c
@@ -363,3 +363,28 @@
 	security_dcon_if0_rr_hbw_config(spi_en);
 	security_dcon_if1_rr_hbw_config(spi_en);
 }
+
+uint32_t security_key_soc_location_offset(uint32_t hash_key_index)
+{
+	uint32_t ret;
+	uint8_t key_soc_base_index;
+	uint8_t key_soc_location_index;
+	uint32_t key_soc_location[] = {
+		mmPSOC_SECURITY_HABANA_KEY_0,
+		mmPSOC_SECURITY_PUBLIC_KEY_0_0,
+		mmPSOC_SECURITY_PUBLIC_KEY_1_0,
+		mmPSOC_SECURITY_PUBLIC_KEY_2_0,
+		mmPSOC_SECURITY_PUBLIC_KEY_3_0
+	};
+
+	/* select which index to choose (haban, customer 0-4) */
+	key_soc_base_index = hash_key_index / SECURITY_DWORD_PER_KEY;
+	/* key is 256bit, 32bit * 8... which of the 8 location is requested */
+	key_soc_location_index = hash_key_index % SECURITY_DWORD_PER_KEY;
+
+	/* get base address of selected key index */
+	ret = key_soc_location[key_soc_base_index];
+	/* add to base address of key the spicifice location of currect DWORD */
+	ret += (key_soc_location_index * sizeof(uint32_t));
+	return ret;
+}
